### ISSUE TITLE
Feature Request: Streamline creation of megaprojects

### ISSUE THREAD

**Issue by sad-dev** [NationalSecurityAgency/ghidra#5551]
**Is your feature request related to a problem? Please describe.**
This F/R originates from discussion #5530.

I have a large (thousands, totalling several GB) set of files that I would like to perform batch analysis on and combine into a 'megaproject', which is particularly nice to have for inter-related files e.g. an entire filesystem. This is currently unfeasible due to the project lock.

**Describe the solution you'd like**
In order of desirability

1.) Functionality to batch analyze a folder in parallel
2.) Headless API to copy programs from one project to another. We can then analyze each file into its own project before copying over to the main project, which allows us to parallelize the analysis portion.

**Describe alternatives you've considered**
GhidraServer?
As mentioned in https://github.com/NationalSecurityAgency/ghidra/discussions/5530#discussioncomment-6420926, UI functionality exists to perform (2)

**Comment by ryanmkurtz** [comment]
We just discussed this and our official recommendation is to run multiple instances of the headless analyzer in parallel, which all point to a GhidraServer shared project.  This will not have any project locking issues.  We discussed your desired solutions, but have no plans to implement them at this time if the server approach works.
